diff --git a/build-tests/x86/suse/test-image-azure/config.sh b/build-tests/x86/suse/test-image-azure/config.sh
index 8218f60954..c15cc8d5ba 100644
--- a/build-tests/x86/suse/test-image-azure/config.sh
+++ b/build-tests/x86/suse/test-image-azure/config.sh
@@ -77,7 +77,7 @@ sed -i -e 's/AutoUpdate.Enabled=y/AutoUpdate.Enabled=n/' /etc/waagent.conf
 
 # Remove the password for root
 # Note the string matches the password set in the config file
-sed -i 's/$1$wYJUgpM5$RXMMeASDc035eX.NbYWFl0/*/' /etc/shadow
+# sed -i 's/$1$wYJUgpM5$RXMMeASDc035eX.NbYWFl0/*/' /etc/shadow
 
 # Implement password policy
 # Length: 6-72 characters long
diff --git a/build-tests/x86/suse/test-image-ec2/config.sh b/build-tests/x86/suse/test-image-ec2/config.sh
index 11f758efdd..6a4387b0eb 100644
--- a/build-tests/x86/suse/test-image-ec2/config.sh
+++ b/build-tests/x86/suse/test-image-ec2/config.sh
@@ -64,7 +64,7 @@ sed -i 's/#ChallengeResponseAuthentication yes/ChallengeResponseAuthentication n
 
 # Remove the password for root
 # Note the string matches the password set in the config file
-sed -i 's/$1$wYJUgpM5$RXMMeASDc035eX.NbYWFl0/*/' /etc/shadow
+# sed -i 's/$1$wYJUgpM5$RXMMeASDc035eX.NbYWFl0/*/' /etc/shadow
 
 #======================================
 # Activate services
diff --git a/build-tests/x86/suse/test-image-gce/config.sh b/build-tests/x86/suse/test-image-gce/config.sh
index 73ec2c3018..79eafa630a 100644
--- a/build-tests/x86/suse/test-image-gce/config.sh
+++ b/build-tests/x86/suse/test-image-gce/config.sh
@@ -72,7 +72,7 @@ sed -i 's/#ChallengeResponseAuthentication yes/ChallengeResponseAuthentication n
 
 # Remove the password for root
 # Note the string matches the password set in the config file
-sed -i 's/$1$wYJUgpM5$RXMMeASDc035eX.NbYWFl0/*/' /etc/shadow
+# sed -i 's/$1$wYJUgpM5$RXMMeASDc035eX.NbYWFl0/*/' /etc/shadow
 
 # Create the boto config file
 echo  '[Boto]
diff --git a/build-tests/x86/suse/test-image-qcow-openstack/config.sh b/build-tests/x86/suse/test-image-qcow-openstack/config.sh
index 07d00c3c52..f381eb7aec 100644
--- a/build-tests/x86/suse/test-image-qcow-openstack/config.sh
+++ b/build-tests/x86/suse/test-image-qcow-openstack/config.sh
@@ -51,7 +51,7 @@ sed -i 's/#ChallengeResponseAuthentication yes/ChallengeResponseAuthentication n
 
 # Remove the password for root
 # Note the string matches the password set in the config file
-sed -i 's/$1$wYJUgpM5$RXMMeASDc035eX.NbYWFl0/*/' /etc/shadow
+# sed -i 's/$1$wYJUgpM5$RXMMeASDc035eX.NbYWFl0/*/' /etc/shadow
 
 # Setup cloud-init for OpenStack
 echo ""  >> /etc/cloud/cloud.cfg
