diff --git a/.bumpversion.cfg b/.bumpversion.cfg
index f62718b89a..e00f05dcfe 100644
--- a/.bumpversion.cfg
+++ b/.bumpversion.cfg
@@ -1,5 +1,5 @@
 [bumpversion]
-current_version = 9.23.0
+current_version = 9.23.1
 commit = True
 tag = True
 
diff --git a/doc/source/conf.py b/doc/source/conf.py
index 1cddaf3ff8..607e6002cc 100644
--- a/doc/source/conf.py
+++ b/doc/source/conf.py
@@ -128,7 +128,7 @@ def setup(app):
 # built documents.
 #
 # The short X.Y version.
-version = '9.23.0'
+version = '9.23.1'
 # The full version, including alpha/beta/rc tags.
 release = version
 
diff --git a/kiwi/version.py b/kiwi/version.py
index 542188f8d6..8072e1a3e6 100644
--- a/kiwi/version.py
+++ b/kiwi/version.py
@@ -18,5 +18,5 @@
 """
 Global version information used in kiwi and the package
 """
-__version__ = '9.23.0'
+__version__ = '9.23.1'
 __githash__ = '$Format:%H$'
