diff --git a/dracut/modules.d/99kiwi-lib/kiwi-partitions-lib.sh b/dracut/modules.d/99kiwi-lib/kiwi-partitions-lib.sh
index 356a16b054..372b2a9918 100644
--- a/dracut/modules.d/99kiwi-lib/kiwi-partitions-lib.sh
+++ b/dracut/modules.d/99kiwi-lib/kiwi-partitions-lib.sh
@@ -179,7 +179,7 @@ function get_partition_node_name {
     local part
     udev_pending
     for partnode in $(
-        lsblk -p -r -o NAME,TYPE "${disk}" | grep -E "part|md$" | cut -f1 -d ' '
+        lsblk -p -l -o NAME,TYPE "${disk}" | grep -E "part|md$" | cut -f1 -d ' '
     );do
         if [ "${index}" = "${partid}" ];then
             echo "${partnode}"
