diff --git a/build-tests/x86/suse/test-image-qcow-openstack/appliance.kiwi b/build-tests/x86/suse/test-image-qcow-openstack/appliance.kiwi
index c183ab9896..774eb68995 100644
--- a/build-tests/x86/suse/test-image-qcow-openstack/appliance.kiwi
+++ b/build-tests/x86/suse/test-image-qcow-openstack/appliance.kiwi
@@ -116,60 +116,7 @@
         <package name="elfutils"/>
         <package name="iputils"/>
         <package name="timezone"/>
-        <package name="yast2"/>
-        <package name="yast2-add-on"/>
-        <package name="yast2-audit-laf"/>
-        <package name="yast2-bootloader"/>
-        <package name="yast2-core"/>
-        <package name="yast2-country"/>
-        <package name="yast2-country-data"/>
-        <package name="yast2-dhcp-server"/>
-        <package name="yast2-dns-server"/>
-        <package name="yast2-firewall"/>
-        <package name="yast2-ftp-server"/>
-        <package name="yast2-hardware-detection"/>
-        <package name="yast2-http-server"/>
-        <package name="yast2-iscsi-client"/>
-        <package name="yast2-iscsi-lio-server"/>
-        <package name="yast2-isns"/>
-        <package name="yast2-kdump"/>
-        <package name="yast2-ldap"/>
-        <package name="yast2-mail"/>
-        <package name="yast2-network"/>
-        <package name="yast2-nfs-client"/>
-        <package name="yast2-nfs-common"/>
-        <package name="yast2-nfs-server"/>
-        <package name="yast2-nis-client"/>
-        <package name="yast2-nis-server"/>
-        <package name="yast2-ntp-client"/>
-        <package name="yast2-online-update"/>
-        <package name="yast2-online-update-frontend"/>
-        <package name="yast2-packager"/>
-        <package name="yast2-pam"/>
-        <package name="yast2-perl-bindings"/>
-        <package name="yast2-pkg-bindings"/>
-        <package name="yast2-printer"/>
-        <package name="yast2-samba-client"/>
-        <package name="yast2-samba-server"/>
-        <package name="yast2-schema"/>
-        <package name="yast2-security"/>
-        <package name="yast2-slp"/>
-        <package name="yast2-slp-server"/>
-        <package name="yast2-squid"/>
-        <package name="yast2-storage-ng"/>
-        <package name="yast2-sudo"/>
-        <package name="yast2-support"/>
-        <package name="yast2-sysconfig"/>
-        <package name="yast2-tftp-server"/>
-        <package name="yast2-transfer"/>
-        <package name="yast2-trans-stats"/>
-        <package name="yast2-tune"/>
-        <package name="yast2-update"/>
-        <package name="yast2-users"/>
-        <package name="yast2-xml"/>
-        <package name="yast2-ycp-ui-bindings"/>
         <package name="docker"/>
-        <package name="yast2-docker"/>
         <package name="cloud-init"/>
         <package name="cloud-init-config-suse"/>
     </packages>
