diff --git a/tools/dcounter.c b/tools/dcounter.c
index 9df56fd81b..42df97447a 100644
--- a/tools/dcounter.c
+++ b/tools/dcounter.c
@@ -9,7 +9,7 @@ int main(int argc, char **argv) {
     const int max_prefix_len = 512;
     char prefix[max_prefix_len];
     unsigned cnt = 0, blk_size = 1024*1024, blk_cnt = 0, size = 0;
-    ssize_t r, p;
+    ssize_t r;
     char buf[1024*1024];
     argv++; argc--;
     memset (prefix,'\0',max_prefix_len);
@@ -30,8 +30,9 @@ int main(int argc, char **argv) {
         return 1;
     }
     fprintf(stderr, "      ");
-    while((r = read(0, buf, sizeof buf)) > 0) {
-        p = 0;
+    r = read(0, buf, sizeof buf);
+    while(r > 0) {
+        ssize_t p = 0;
         ssize_t w = 0;
         while((w = write(1, buf + p, r - p)) >= 0) {
             p += w;
@@ -57,6 +58,7 @@ int main(int argc, char **argv) {
                 }
             }
         }
+        r = read(0, buf, sizeof buf);
     }
     fflush(stdout);
     return 0;
