diff --git a/build-tests/x86/archlinux/test-image-live-disk-kis/appliance.kiwi b/build-tests/x86/archlinux/test-image-live-disk-kis/appliance.kiwi
index 0dff3800f7..863977ee29 100644
--- a/build-tests/x86/archlinux/test-image-live-disk-kis/appliance.kiwi
+++ b/build-tests/x86/archlinux/test-image-live-disk-kis/appliance.kiwi
@@ -31,12 +31,12 @@
         <rpm-check-signatures>false</rpm-check-signatures>
     </preferences>
     <preferences profiles="Live">
-        <type image="iso" flags="overlay" hybridpersistent_filesystem="ext4" hybridpersistent="true" firmware="efi" kernelcmdline="console=ttyS0">
+        <type image="iso" flags="overlay" hybridpersistent_filesystem="ext4" hybridpersistent="true" kernelcmdline="console=ttyS0">
             <bootloader name="grub2" console="serial"/>
         </type>
     </preferences>
     <preferences profiles="Virtual">
-        <type image="oem" primary="true" filesystem="ext4" firmware="efi" kernelcmdline="console=ttyS0">
+        <type image="oem" primary="true" filesystem="ext4" firmware="efi" kernelcmdline="console=ttyS0" editbootinstall="editbootinstall_arch.sh">
             <oemconfig>
                 <oem-resize>false</oem-resize>
             </oemconfig>
@@ -44,7 +44,13 @@
         </type>
     </preferences>
     <preferences profiles="Disk">
-        <type image="oem" filesystem="ext4" installiso="true" kernelcmdline="console=ttyS0">
+        <!--
+             KIWI does not natively provide EFI support for Arch Linux,
+             because of that the editbootinstall script is used to hot pacth
+             grub.cfg file. This hotpatch only applies for disk images
+             therefore the installation ISO does not support EFI
+        -->
+        <type image="oem" filesystem="ext4" installiso="true" kernelcmdline="console=ttyS0" editbootinstall="editbootinstall_arch.sh" firmware="efi">
             <bootloader name="grub2" console="serial"/>
         </type>
     </preferences>
diff --git a/build-tests/x86/archlinux/test-image-live-disk-kis/editbootinstall_arch.sh b/build-tests/x86/archlinux/test-image-live-disk-kis/editbootinstall_arch.sh
new file mode 100644
index 0000000000..725d2822cc
--- /dev/null
+++ b/build-tests/x86/archlinux/test-image-live-disk-kis/editbootinstall_arch.sh
@@ -0,0 +1,22 @@
+#!/bin/bash -x
+
+# Arch Linux does not support linuxefi grub module or command,
+# however KIWI relies on it to create a grub.cfg that can be
+# used for EFI and non EFI environments.
+# In Arch linux command is already capable to EFI boot
+set -e
+
+disk_img=$1
+mapped_dev=$2
+tmp_mount=$(mktemp -d disk_XXXXX)
+
+mount "${mapped_dev}" "${tmp_mount}"
+
+grub_config="${tmp_mount}/boot/grub/grub.cfg"
+
+if [ -f  "${grub_config}" ]; then
+    echo "Patching grub configuration file: ${grub_config}"
+    sed -i "s|linuxefi|linux|g" "${grub_config}"
+    sed -i "s|initrdefi|initrd|g" "${grub_config}"
+fi
+umount "${tmp_mount}" && rm -r ${tmp_mount}
