diff --git a/package/python-kiwi-spec-template b/package/python-kiwi-spec-template
index 0d4eba7932..315e829aa1 100644
--- a/package/python-kiwi-spec-template
+++ b/package/python-kiwi-spec-template
@@ -127,7 +127,7 @@ Provides:       kiwi-image-wsl-requires = %{version}-%{release}
 Obsoletes:      kiwi-image-wsl-requires < %{version}-%{release}
 %if "%{_vendor}" != "debbuild"
 Provides:       kiwi-image:docker
-Provides:       kiwi-image:wsl
+Provides:       kiwi-image:appx
 %endif
 %if 0%{?suse_version}
 Requires:       umoci
