diff --git a/.bumpversion.cfg b/.bumpversion.cfg
index 3e49833907..2ff0a265ab 100644
--- a/.bumpversion.cfg
+++ b/.bumpversion.cfg
@@ -1,9 +1,8 @@
 [bumpversion]
-current_version = 9.21.26
+current_version = 9.22.0
 commit = True
 tag = True
 
 [bumpversion:file:kiwi/version.py]
 
 [bumpversion:file:doc/source/conf.py]
-
diff --git a/doc/source/conf.py b/doc/source/conf.py
index 1364aaa3fa..a142a41f92 100644
--- a/doc/source/conf.py
+++ b/doc/source/conf.py
@@ -128,7 +128,7 @@ def setup(app):
 # built documents.
 #
 # The short X.Y version.
-version = '9.21.26'
+version = '9.22.0'
 # The full version, including alpha/beta/rc tags.
 release = version
 
diff --git a/kiwi/version.py b/kiwi/version.py
index 8e7777f900..4af9a4ce3e 100644
--- a/kiwi/version.py
+++ b/kiwi/version.py
@@ -18,5 +18,5 @@
 """
 Global version information used in kiwi and the package
 """
-__version__ = '9.21.26'
+__version__ = '9.22.0'
 __githash__ = '$Format:%H$'
