diff --git a/kiwi/container/__init__.py b/kiwi/container/__init__.py
index c64031614b..924f72aed0 100644
--- a/kiwi/container/__init__.py
+++ b/kiwi/container/__init__.py
@@ -20,6 +20,7 @@
     ABCMeta,
     abstractmethod
 )
+from typing import Dict
 
 from kiwi.exceptions import (
     KiwiContainerImageSetupError
@@ -39,7 +40,7 @@ def __init__(self) -> None:
         return None  # pragma: no cover
 
     @staticmethod
-    def new(name: str, root_dir: str, custom_args: dict = None):
+    def new(name: str, root_dir: str, custom_args: Dict=None):  # noqa: E252
         name_map = {
             'docker': 'OCI',
             'oci': 'OCI',
diff --git a/kiwi/container/setup/__init__.py b/kiwi/container/setup/__init__.py
index f84b4dfaf9..4e918b11be 100644
--- a/kiwi/container/setup/__init__.py
+++ b/kiwi/container/setup/__init__.py
@@ -20,6 +20,7 @@
     ABCMeta,
     abstractmethod
 )
+from typing import Dict
 
 # project
 from kiwi.exceptions import (
@@ -36,7 +37,7 @@ def __init__(self) -> None:
         return None  # pragma: no cover
 
     @staticmethod
-    def new(name: str, root_dir: str, custom_args: dict = None):
+    def new(name: str, root_dir: str, custom_args: Dict=None):  # noqa: E252
         name_map = {
             'docker': 'Docker',
             'oci': 'OCI',
